### PR TITLE
Re-factor

### PR DESCRIPTION
**Pull request by zeroslack**
Cleans up some style, re-structures, and adds testing infrastructure w/minimal tests.

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -1,13 +1,14 @@
...
+    - 3.5
     - 3.6
 install:
     - sudo apt-get -y update
     - sudo apt-get -y install python-matplotlib
-    - pip install codecov
+    - pip install -e .
+    - pip install -r test-requirements.txt
 script:
-    - nosetests
```
Comment (evanbiederstedt) [inline .travis.yml]: @kamidzi You don't think nosetests are still worthwhile?

File: setup.py [hunk 2]
```diff
@@ -45,7 +50,7 @@ def _get_authors_str():
         'Operating System :: POSIX :: Linux',
         'Programming Language :: Python :: 3 :: Only',
         'Programming Language :: Python :: 3',
-        'Programming Language :: Python :: 3.4',
+        'Programming Language :: Python :: 3.5',
         'Programming Language :: Python :: 3.6',
         'Programming Language :: Python :: 3.7',
```
Comment (evanbiederstedt) [inline setup.py]: @kamidzi Is 3.7 considered stable? I'm not sure it would be for all dependencies...

**Review by evanbiederstedt — APPROVED** [review]
We should probably compress promoters.hg19 and the VCF. Otherwise, so far lgtm

File: .travis.yml (hunk 1 above)
Comment (zeroslack) [reply .travis.yml]: I've replaced `nose` with `tox` and `pytest`. With travis, only `pytest` is executed. Locally, one could call `tox` (which also supports `nose`) to execute `pytest` for all the defined python versions, or simply execute `pytest` to test just the currently configured python version... I'll add a TESTING section to the README with this info.

**Comment by kamidzi** [comment]
This project has been dead, so might as well merge this. :(

**Comment by evanbiederstedt** [comment]
Thanks for merging! @kamidzi Apologies that it's been on the back burner

**Pull request merged**